tools/Dashboard/DevHome.Dashboard/Helpers/WidgetServiceHelper.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using Windows.Management.Deployment;
+
+namespace DevHome.Dashboard.Helpers;
+internal class WidgetServiceHelper
+{
+    private readonly Version minSupportedVersion400 = new (423, 3800);
+    private readonly Version minSupportedVersion500 = new (523, 3300);
+    private readonly Version version500 = new (500, 0);
+
+    private bool _validatedWebExpPack;
+
+    public WidgetServiceHelper()
+    {
+        _validatedWebExpPack = false;
+    }
+
+    public bool EnsureWebExperiencePack()
+    {
+        // If already validated there's a good version, don't check again.
+        if (_validatedWebExpPack)
+        {
+            return true;
+        }
+
+        // Ensure the application is installed, and the version is high enough.
+        const string packageName = "MicrosoftWindows.Client.WebExperience_cw5n1h2txyewy";
+
+        var packageManager = new PackageManager();
+        var packages = packageManager.FindPackagesForUser(string.Empty, packageName);
+        if (packages.Any())
+        {
+            // A user cannot actually have more than one version installed, so only need to look at the first result.

```

Comment:
Why can't a user have more then one version of WebExpirence installed?

Reply:
Is the WebExpirence package required for DevHome?

Reply:
It is required for widgets. If someone tries to add a widget, and it's not installed or not a new enough version, we present a link to the store.

Reply:
If it is needed, can the package dependency be added to the MSIX for DevHome?

Reply:
When we started going down that road, it got complicated quickly, but it probably makes sense to open a work item for investigating that.

Reply:
This was really always going to be a temporary solution, since if we can go down-level to Win10 we need a whole thing for that.